eng/pipelines/templates/jobs/build.yml

```diff
@@ -76,6 +76,65 @@ jobs:
     inputs:
       summaryFileLocation: $(CoverageFile)
 
+  # AOT Compatibility Analysis - only run on Linux x64 to avoid redundancy
+  - task: PowerShell@2
+    displayName: "Generate AOT Compatibility Report"
+    condition: and(succeeded(), eq('${{ parameters.OSName }}', 'linux'), eq(variables['Architecture'], 'x64'))
+    inputs:
+      pwsh: true
+      filePath: $(Build.SourcesDirectory)/eng/scripts/Generate-AOTCompactReport.ps1
+      arguments: >
+        -Runtime 'linux-x64'
+        -GenerateHtml:$false
+      workingDirectory: $(Build.SourcesDirectory)
+    continueOnError: true
+
+  - task: PowerShell@2
+    displayName: "Display AOT Analysis Report"
+    condition: and(always(), eq('${{ parameters.OSName }}', 'linux'), eq(variables['Architecture'], 'x64'))
+    inputs:
+      pwsh: true
+      targetType: 'inline'
+      script: |
+        $jsonPath = "$(Build.SourcesDirectory)/.work/aotCompactReport/aot-compact-report.json"
+        
+        if (Test-Path $jsonPath) {
+          $jsonContent = Get-Content $jsonPath -Raw
+          $jsonObject = $jsonContent | ConvertFrom-Json
+          $totalWarnings = 0
+          $dllCount = 0
+          
+          if ($jsonObject -is [PSCustomObject]) {
+            foreach ($property in $jsonObject.PSObject.Properties) {
+              $dllCount++
+              $totalWarnings += $property.Value.Count
+            }
+          }
+          
+          Write-Host "##[section]AOT Compatibility Analysis Results"
+          Write-Host "Total AOT/Trimming Warnings: $totalWarnings"
+          Write-Host "Affected DLLs: $dllCount"
+          Write-Host ""
+          Write-Host "##[section]Full AOT Analysis Report (JSON):"
+          Write-Host $jsonContent
```

Comment:
In CI we show AOT warns in JSON format (as shown in the pr description). we can think of rendering the html report, but not in the scope of this pr, json is readable enough atm.